Simplenote/Information/NoteInformationController.swift

```diff
@@ -53,23 +95,37 @@ final class NoteInformationController {
 private extension NoteInformationController {
     func startListeningForChanges() {
         noteChangesObserver.delegate = self
+
+        referencesController?.onDidChangeContent = { [weak self] _, _ in
+            self?.sendNewDataToObserver()
+        }
     }
 
     func stopListeningForChanges() {
         noteChangesObserver.delegate = nil
+        referencesController?.onDidChangeContent = nil
     }
 }
 
 // MARK: - Data
 //
 private extension NoteInformationController {
-    func allRows() -> [Row] {
-        return metricRows()
+    func allSections() -> [Section] {
+        var sections: [Section] = []
+        sections.append(metricSection())
+        if let referenceSection = referenceSection() {
```

Comment:
Nipticky: switching to `guard let ... else` would let you drop the indentation level below 🔥